table/table.cpp

```diff
@@ -275,11 +307,29 @@ void Table::output_entry(::database::Entry& entry) {
 Table::Info Table::column_index(const string &column) {
     auto it = column_indecies.find(column);
     if(it == column_indecies.end()) {
-        throw out_of_range("Column Does Not Exist!");
+        throw out_of_range("Column '" + column + "' Does Not Exist!");
     }
     return it->second;
 } // column_index()
 
+// Loops through all comparisons, checking if row entry@index is equal to the corresponding comparison
+// If all checks are good, returns true; else, returns false
+bool Table::check_row(const ::database::Row &row, vector<Info> indecies, vector<::database::Entry> &comparisons) {
+    bool good = true;
+    for(size_t j = 0; j < comparisons.size(); ++j) {
+        bool correct_comparison = correct_type(indecies[j], comparisons[j]);
+        if(!correct_comparison) {
+            throw type_mismatch(type_mismatch::error_type::comparison, get_type(indecies[j]), get_type(comparisons[j]));
+        }
+        
+        if(!compare_entries(row.entries(indecies[j].index), comparisons[j])) {
+            good = false;
+            break;
+        }
+    }
+    return good;
+}
+
 bool Table::compare_entries(const ::database::Entry &lhs, const ::database::Entry &rhs) {
```

Comment:
Not required for this PR, but this method is confusing to read. The word "compare" implies there is some condition being checked, but it looks like you are just checking if the value of the entries are equal.

Correct me if I'm wrong, but for two protobuf objects, you can just check if lhs == rhs to assert equality, without needing to assert that their individual fields both exist and are the same value. So this method just adds complexity to the code.

In a separate PR, consider removing this method and just using == operator where this function is used.